src/gamecontroller.cs

```diff
@@ -535,4 +621,1162 @@ interface GCEventViewController {
 		[Export ("controllerUserInteractionEnabled")]
 		bool ControllerUserInteractionEnabled { get; set; }
 	}
+
+	[TV (14,0), Mac (11,0), iOS (14,0)]
+	[BaseType (typeof( NSObject))]
+	[DisableDefaultCtor]
+	interface GCColor : NSCopying, NSSecureCoding
+	{
+		[Export ("initWithRed:green:blue:")]
+		IntPtr Constructor (float red, float green, float blue);
+
+		[Export ("red")]
+		float Red { get; }
+
+		[Export ("green")]
+		float Green { get; }
+
+		[Export ("blue")]
+		float Blue { get; }
+	}
+
+	delegate void GCControllerTouchpadHandler (GCControllerTouchpad touchpad, float xValue, float yValue, float buttonValue, bool buttonPressed);
+
+	[TV (14,0), Mac (11,0), iOS (14,0)]
+	[BaseType (typeof (GCControllerElement))]
+	interface GCControllerTouchpad
+	{
+		[Export ("button")]
+		GCControllerButtonInput Button { get; }
+
+		[NullAllowed, Export ("touchDown", ArgumentSemantic.Copy)]
+		GCControllerTouchpadHandler TouchDown { get; set; }
+
+		[NullAllowed, Export ("touchMoved", ArgumentSemantic.Copy)]
+		GCControllerTouchpadHandler TouchMoved { get; set; }
+
+		[NullAllowed, Export ("touchUp", ArgumentSemantic.Copy)]
+		GCControllerTouchpadHandler TouchUp { get; set; }
+
+		[Export ("touchSurface")]
+		GCControllerDirectionPad TouchSurface { get; }
+
+		[Export ("touchState")]
+		GCTouchState TouchState { get; }
+
+		[Export ("reportsAbsoluteTouchSurfaceValues")]
+		bool ReportsAbsoluteTouchSurfaceValues { get; set; }
+
+		[Export ("setValueForXAxis:yAxis:touchDown:buttonValue:")]
+		void SetValue (float xAxis, float yAxis, bool touchDown, float buttonValue);
+	}
+
+	[TV (14,0), Mac (11,0), iOS (14,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface GCDeviceBattery
+	{
+		[Export ("batteryLevel")]
+		float BatteryLevel { get; }
+
+		[Export ("batteryState")]
+		GCDeviceBatteryState BatteryState { get; }
+	}
+
+	[TV (14,0), Mac (11,0), iOS (14,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface GCDeviceHaptics
+	{
+		[Export ("supportedLocalities", ArgumentSemantic.Strong)]
+		NSSet<NSString> SupportedLocalities { get; }
+
+		// Dependency on Xcode 12 CoreHaptics bindings
```

Comment:
FYI @mandel-macaque.

Reply:
I know this guy. It is in my feedback report, thanks for the headsup :)

Reply:
Do we want to link to the issue here for easier record keeping? :)

Reply:
Looks like CoreHaptics is blocked on AVAudioSession for Mac bindings :(

https://github.com/xamarin/maccore/issues/2261